src/basic_bot/commons/hub_state_monitor.py

```diff
@@ -12,6 +12,10 @@
 from basic_bot.commons.hub_state import HubState
 
 
+# TODO: This class should maybe be a singleton.
+should_exit = False
```

Comment:
I loathe, but there doesn't seem to be another reliable way to kill a python 3.9 thread?   Tried using an instance variable, but I think maybe python is making a copy of the class state when the thread is created.